src/net/mod.rs

```diff
@@ -838,6 +839,19 @@ pub struct NatPunchData {
     pub nonce: u32,
 }
 
+#[repr(u8)]
+#[derive(Debug, Clone, PartialEq)]
+pub enum MemPoolSyncDataID {
+    BloomFilter = 0x01,
+    TxTags = 0x02,
+}
```

Comment:
This should use the `define_u8_enum`. It provides a `to_u8()` method for variants, and a `from_u8(v: u8) -> Option<Self>` method. That avoids needing to do cast-matches, because you can check the result of `from_u8`, and then match on the enum, which allows the compiler to enforce that all variants are handled.

Reply:
Done